algorithm-prototyping/brute-force-prototype/src/main/java/edu/uga/devdogs/bruteforceprototype/schedule/ScheduleUtil.java

```diff
@@ -55,7 +55,26 @@ public static double computeMaxDistance(Schedule schedule, Distances distances)
      * @return the average idle time between classes in the schedule
      */
     public static double computeAverageIdleTime(Schedule schedule) {
-        return 0.0;
+        int countOfClassGaps = 0;
+        int sumOfTimeGaps = 0;
+
+        // Loop through the days
+        for (TreeSet<Class> day: schedule.days().values()){
+            int lastTimeEnd = -1;
+            // Loop through all of the classes in a single day
+            for (Class iClass: day){
+                if (lastTimeEnd == -1){
+                    lastTimeEnd = iClass.endTime().getHour()*60 + iClass.endTime().getMinute();
+                    continue;
+                }
+                sumOfTimeGaps += iClass.startTime().getHour()*60 + iClass.startTime().getMinute();
+                sumOfTimeGaps -= lastTimeEnd;
+                lastTimeEnd = iClass.endTime().getHour()*60 + iClass.endTime().getMinute();
+                countOfClassGaps++;
+            }
+        }
+
+        return sumOfTimeGaps / sumOfTimeGaps;
```

Comment:
The current return statement will always return 1 if `sumOfTimeGaps` is non-zero. You probably meant to type `return sumOfTimeGaps / countOfClassGaps`. You should also consider edge cases such if `countOfClassGaps == 0`. You need to handle that to avoid division by zero.

Reply:
There may also be other edge cases. For example, empty schedules, one class per day, and classes with overlapping times. Make sure the method behaves predictably and reasonably for all edge cases.